### PR TITLE
refactor:pass ctx to db layer, return err instead of recover panic

### PR DESCRIPTION
**Pull request by hulb**
This PR makes a refactor of ways to handle error in db layer. I don't think it's proper to panic when there is an error in db actions. And every db action should be executed with a context passed by the caller.

### CODE REVIEW

**Review by fmartingr — COMMENTED** [review]
Hey @hulb, thanks a lot for this (and all the community work!). Sorry it took me so long to review it.

Most of the PR seems good, I believe you didn't break any logic regarding the databases (but made me realize how much repeated code we have and how good an ORM would be here...).

My only "concern" here is the errors library, I have been taking a look at what it proposes and so on, but I don't know if we should implement this now regarding the [future error handling go will propose](https://go.googlesource.com/proposal/+/master/design/go2draft-error-handling-overview.md).

In order to wrap errors in go, we can use `fmt.Errorf("somethihg happened: %w", err)` and then just `Unwrap` until we find what we want. Or implement our own error data structure, maybe, but using `runtime.Callers` doesn't seem like good practice.

What do you think? Maybe this makes development really easy and I'm just not seeing it, I want your feedback on that.

File: go.mod [hunk 1]
```diff
@@ -17,10 +17,12 @@ require (
 	github.com/lib/pq v1.10.5
 	github.com/muesli/go-app-paths v0.2.2
 	github.com/patrickmn/go-cache v2.1.0+incompatible
+	github.com/pkg/errors v0.9.1
```
[the remark this answers is not in the document]
Comment (hulb) [reply go.mod:20]: Actually I intended to use `github.com/pkg/errors` here to catch a err with stack trace to let the uppper caller know where the err exactly occures. In my experience of work, the `github.com/pkg/errors ` is now the de-factor standard of err handling. It implements more convenient error wrapping and unwrapping. It's some kind of habit of mine to use the error package.

As for the draft proposal, I think it's a long way. If a better err handling comes out in the future, we can migrate to it.

File: internal/cmd/root.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/go-shiori/shiori/internal/database"
 	apppaths "github.com/muesli/go-app-paths"
 	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
```
[the remark this answers is not in the document]
Comment (hulb) [reply internal/cmd/root.go:11]: Yes, my mistake

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (hulb) [reply go.mod:20]: Is there some drawback of using `runtime.Callers` ? I don't really know it very well. The reason I use `github.com/pkg/errors` is just to wrap a trace stack in err

**Comment by hulb** [comment]
@fmartingr I think I need more time to test this PR in case of introducing bugs. I have been a little busy these days. When I finish my test I'll let you know.

**Comment by hulb** [comment]
And I think it's so complicated in some implemention of db actions, like `SaveBookmarks`. There are so many codes and logic there. This PR the first step for me to refine these blocks.

**Comment by fmartingr** [comment]
Thank you @hulb , let's go ahead with this one and figure out things one step at a time :)

**Pull request merged**